services/dqa-secondary-reports/main.py

```diff
@@ -257,27 +268,63 @@ def index():
             'uuid': SERVICE_ID,
         })
 
-    app.add_url_rule('/pedsnet',
+    """
+    Make sure all urls end with a slash. The version without a trailing slash
+    will be taken care of automatically because Flask will automatically
+    redirect the user to the same URL with a trailing slash attached.
+    The reverse is not true.
+    """
+
+    app.add_url_rule('/pedsnet/',
                      'pedsnet_versions',
                      dqa_resource.serve_versions,
                      methods=['GET'])
 
     for version in dqa_resource.get_versions():
-        app.add_url_rule('/pedsnet/' + version + '/field-totals',
+        app.add_url_rule('/pedsnet/' + version + '/field-totals/',
                          version + '_' + 'field-totals',
                          dqa_resource.serve_field_totals(version),
                          methods=['GET'])
 
-        app.add_url_rule('/pedsnet/' + version + '/site-totals',
+        for table in dqa_resource.field_totals[version]:
+            app.add_url_rule(('/pedsnet/' + version + '/field-totals/' +
+                              table.replace('_', '-') + '/'),
```

Comment:
You're going to hate me for this, but for identifiers as parameters it is perfectly fine to leave the underscores. I know said the convention is hyphens, but that is for static segments of the path. For parameters, it is more clear leaving them as is.